proto/desmos/posts/v3/models.proto

```diff
@@ -0,0 +1,342 @@
+  string url = 3 [ (gogoproto.moretags) = "yaml:\"url\"" ];
+
+  // (optional) Display value of the URL
+  string display_url = 4 [ (gogoproto.moretags) = "yaml:\"display_url\"" ];
+}
+
+// Attachment contains the data of a single post attachment
+message Attachment {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  // Id of the subspace inside which the post to which this attachment should be
+  // connected is
+  uint64 subspace_id = 1 [
+    (gogoproto.customname) = "SubspaceID",
+    (gogoproto.moretags) = "yaml:\"subspace_id\""
+  ];
+
+  // Id of the post to which this attachment should be connected
+  uint64 post_id = 2 [
+    (gogoproto.customname) = "PostID",
+    (gogoproto.moretags) = "yaml:\"post_id\""
+  ];
+
+  // If of this attachment
+  uint32 id = 3
+      [ (gogoproto.customname) = "ID", (gogoproto.moretags) = "yaml:\"id\"" ];
+
+  // Content of the attachment
+  google.protobuf.Any content = 4 [
+    (cosmos_proto.accepts_interface) = "AttachmentContent",
+    (gogoproto.moretags) = "yaml:\"content\""
+  ];
+}
+
+// Media represents a media attachment
+message Media {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+  option (cosmos_proto.implements_interface) = "AttachmentContent";
+
+  string uri = 2 [ (gogoproto.moretags) = "yaml:\"uri\"" ];
+  string mime_type = 3 [ (gogoproto.moretags) = "yaml:\"mime_type\"" ];
+}
+
+// Poll represents a poll attachment
+message Poll {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+  option (cosmos_proto.implements_interface) = "AttachmentContent";
+
+  // Question of the poll
+  string question = 1 [ (gogoproto.moretags) = "yaml:\"question\"" ];
+
+  // Answers the users can choose from
+  repeated ProvidedAnswer provided_answers = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"provided_answers\""
+  ];
+
+  // Date at which the poll will close
+  google.protobuf.Timestamp end_date = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.stdtime) = true,
+    (gogoproto.moretags) = "yaml:\"end_date\""
+  ];
+
+  // Whether the poll allows multiple choices from the same user or not
+  bool allows_multiple_answers = 4
+      [ (gogoproto.moretags) = "yaml:\"allows_multiple_answers\"" ];
+
+  // Whether the poll allows to edit an answer or not
+  bool allows_answer_edits = 5
+      [ (gogoproto.moretags) = "yaml:\"allows_answers_edits\"" ];
+
+  // Final poll results
+  PollTallyResults final_tally_results = 6
+      [ (gogoproto.moretags) = "yaml:\"final_tally_results\"" ];
+
+  // Provided answer contains the details of a possible poll answer
+  message ProvidedAnswer {
+    option (gogoproto.equal) = true;
+    option (gogoproto.goproto_stringer) = true;
+
+    // (optional) Text of the answer
+    string text = 1 [ (gogoproto.moretags) = "yaml:\"text\"" ];
+
+    // Content of the attachment
+    repeated google.protobuf.Any attachments = 2 [
```

Comment:
I would name it `contents`.

Reply:
I've left it named `attachments` since it makes more sense. Even though the type is `AttachmentContent`, the list still identifies all the attachments associated with a specific provided answer. Ithink naming it `contents` would confuse users: what is the _content_ of a provided answer? The answer itself? Leaving it `attachments` instead makes it easy to understand in my opinion